includes/wf_crm_webform_postprocess.inc

```diff
@@ -240,6 +240,26 @@ class wf_crm_webform_postprocess extends wf_crm_webform_base {
     if (empty($this->submission->is_draft) && !empty($this->ent['contribution'][1]['id']) && $this->contributionIsIncomplete && !$this->contributionIsPayLater) {
       $this->submitIPNPayment();
     }
+      // Send receipt 
+    if (empty($this->submission->is_draft) 
+        && !empty($this->ent['contribution'][1]['id']) 
+        && !empty($this->contribution_page['is_email_receipt']) 
+        && (!$this->contributionIsIncomplete || $this->contributionIsPayLater) ) {
+      $this->sendReceipt();
+    }
+  }
+
+  /**
+   * Send receipt 
+   */
+  private function sendReceipt(){
+    // tax integration
+    if (!is_null($this->tax_rate)) {
+      $template = CRM_Core_Smarty::singleton();
+      $template->assign('dataArray', array( "{$this->tax_rate}" => $this->tax_rate/100 ));
+    }
+    $contribute_id = $this->ent['contribution'][1]['id'];
+    wf_civicrm_api('contribution', 'sendconfirmation', array('id' => $contribute_id) + $this->contribution_page);
```

Comment:
Ok reading the code, this will send a reciept only if:

- it is not an ipn payment
- the contribution record was created
- the contribution page is set to do so
- the contribution is complete OR it is pay later

I'm unsure about that last condition - does sending a receipt for a pay-later transaction make sense? Maybe it does, just wanted to double-check with you and make sure it's what you meant.